www/date.js

```diff
@@ -47,18 +52,35 @@ var date = {
             return error;
       }
    },
-   _startUpdateErrorInterval: function() { 
-      if(!date._updateErrorInterval) {
-         date._updateErrorInterval = window.setInterval(() => {
-            date.update(() => {
-               window.clearInterval(date._updateErrorInterval);
-               date._updateErrorInterval = null;
-            }, date._logger.log);
-         }, TIME_SYNC_ERROR_DELAY);
+   _startInitErrorInterval: function() {
+      if (!date._initErrorInterval) {
+         date._initErrorInterval = window.setInterval(() => {
+            new Promise((resolve, reject) => {
+               date._reinit(resolve, reject);
+            }).then(() => {
+               return new Promise((resolve, reject) => {
+                  date.update(resolve, reject);
+               });
+            }).then(() => {
+               window.clearInterval(date._initErrorInterval);
+               date._initErrorInterval = null;
+            }).catch(date._logger.log);
+         }, INIT_ERROR_INTERVAL_DELAY);
```

Comment:
In my experience using asynchronous methods like this in a loop can be dangerous, particularly if/when the asynchronous action takes longer than the interval time, which tends starts to cause something like an infinite loop, where another interval files while the first interval is still processing through it's asynchronous chain and if this keeps on occurring, over time the JS engine will become so busy handling a never-ending stream of interval callbacks that the app will lock up even if you call on `clearInterval`.

I assume that this code uses network requests, so it is very plausible that this asynchronous chain could take longer than the 10 second interval which can trigger this kind of behaviour.

Instead, it might be better to use a `setTimeout` to trigger a retry on `catch`.

Reply:
Hmmm... So basically, if it fails retry in X milliseconds. If it succeeds, it's done.

Reply:
Right.

Reply:
Alright, can you check the new changes.